source/MaterialXRuntime/RtStage.cpp

```diff
@@ -151,20 +151,16 @@ RtPrim RtStage::createNodeDef(RtPrim nodegraphPrim,
         throw ExceptionRuntimeError("Cannot create nodedef '" + nodeDefName.str() + "', with node name: '" + nodeName.str() + "'");
     }
 
-    // Always used qualified namespace
-    const bool isNameSpaced = !namespaceString.empty();
-    const RtString qualifiedNodeDefName = isNameSpaced ? RtString(namespaceString.str() + MaterialX::NAME_PREFIX_SEPARATOR + nodeDefName.str()) : nodeDefName;
-
     PvtStage* stage = PvtStage::cast(this);
 
     // Make sure the nodedef name is unique among all prims in the stage.
     PvtPath path(nodeDefName);
     if (stage->getPrimAtPath(path))
     {
-        throw ExceptionRuntimeError("The nodedef name '" + qualifiedNodeDefName.str() + "' is not unique");
+        throw ExceptionRuntimeError("The nodedef name '" + nodeDefName.str() + "' is not unique");
```

Comment:
Just double checking, this fixes the issue of adding the namespace prefix when creating the nodedef ?

The assumption here is that the nodedefName is created without any namespacing and the "namespace" string
still needs to be used. 

Thus I think the only change here is that instead of the NAME_PREFIX_SEPERATOR ":" we want to use "_".

Reply:
As part of this we should also fix up the nodegraph name to use the namespace string for uniqueness.

Reply:
Here I am providing the fully qualified Node Definition name, including:
1) ND_ prefix
2) namespace
3) node name
4) output types
5) version

The final qualified node name is build using the PublishCompoundDilog. So, you want to actually append the namespace in this function?